### ISSUE TITLE
Missing extension point to supply other implementations of IStackTraceParser

### ISSUE THREAD

**Issue by mdavidsen** [microsoft/sourcemap-toolkit#67]
Hi,

While trying to use the library to handle other types of stacktraces, specificly in the format Bugsnag for ReactNative supplies the stacktraces:

```json
[
  {
    "function": "onerror",
    "line": 109,
    "column": 2655,
    "filename": "main.jsbundle"
  },
  {
    "function": "value",
    "line": 94,
    "column": 1505,
    "filename": "main.jsbundle"
  },
  {
    "function": "value",
    "line": 93,
    "column": 7697,
    "filename": "main.jsbundle"
  },
  {
    "function": "value",
    "line": 93,
    "column": 4437,
    "filename": "main.jsbundle"
  },
  {
    "function": "value",
    "line": 31,
    "column": 1524,
    "filename": "main.jsbundle"
  },
  {
    "function": "value",
    "line": 19,
    "column": 1592,
    "filename": "main.jsbundle"
  },
  {
    "function": "value",
    "line": 0,
    "column": 0,
    "filename": "[native code]"
  }
]
``` 

I encountered issues with the parser and wanted to extend the parser, but have no way to supply our own implementation of the parser that I can see, without using tons of reflection to extend the `StackTraceDeminfierFactory.GetStackTraceDeminfier()` or directly on the constructor of  `StackTraceDeminifier`. 

How should I go about doing this the proper way? Creating a overload on the Factory-method that allows a user to pass in a implementation of `StackTraceParser` seems like the proper way to go about doing it to me?

**Comment by christiango** [comment]
An overload sounds like a good way to go about it

**Comment by mdavidsen** [comment]
I've open PR #68 to add overloads to the factories

**Comment by christiango** [comment]
Closed with #68
